File: src/LightNap.Core/Identity/Dto/Request/ExternalLoginRegisterRequestDto.cs
using LightNap.Core.Configuration;
using System.ComponentModel.DataAnnotations;

namespace LightNap.Core.Identity.Dto.Request
{
    /// <summary>
    /// Represents the data transfer object for completing external login registration.
    /// </summary>
    public class ExternalLoginRegisterRequestDto : ExternalLoginRequestDto
    {
        /// <summary>
        /// Gets or sets the email address of the user.
        /// </summary>
        [Required]
        [EmailAddress]
        [StringLength(Constants.Dto.MaxUserNameLength)]
        public required string Email { get; set; }

        /// <summary>
        /// Gets or sets the username of the user.
        /// </summary>
        [Required]
        [StringLength(Constants.Dto.MaxUserNameLength)]
        public required string UserName { get; set; }

        /// <summary>
        /// Whether the user is opting into marketing email at registration.
        /// </summary>
        public bool MarketingOptIn { get; set; }
    }
}

File: src/lightnap-ng/src/app/core/backend-api/dtos/external-logins/request/external-login-register-request-dto.ts
import { ExternalLoginRequestDto } from "./external-login-request-dto";

/**
 * Represents a request to register a new user.
 */
export interface ExternalLoginRegisterRequestDto extends ExternalLoginRequestDto {
    /**
     * The username of the user.
     */
    userName: string;

    /**
     * The email address of the user.
     */
    email: string;
}
